cmd/crossplane/core/core.go

```diff
@@ -102,6 +103,11 @@ type startCommand struct {
 	MaxReconcileRate int           `default:"100" help:"The global maximum rate per second at which resources may checked for drift from the desired state."`
 
 	WebhookEnabled bool `default:"true" env:"WEBHOOK_ENABLED" help:"Enable webhook configuration."`
+	WebhookPort    int  `default:"9443" env:"WEBHOOK_PORT"    help:"The port the webhook server listens on."`
+
+	MetricsBindAddress string `default:":8080" env:"METRICS_BIND_ADDRESS" help:"The address the metrics server listens on." placeholder:"host:port"`
+
+	HealthProbeBindAddress string `default:":8081" env:"HEALTH_PROBE_BIND_ADDRESS" help:"The address the health probe endpoint binds to." placeholder:"host:port"`
```

Comment:
it's feeling a bit odd to expose these as addresses in the command line args, but as port numbers in the helm chart values. What do you think about consolidating them and only exposing as port numbers everywhere?

Is there a use case where the user would need to set the host also here in addition to the port? i'm not sure there is, but if there is, we'd probably need to make a change to the helm chart values later anyways to make that easier to set since we aren't exposing it there now.

My instinct is to just expose port numbers everywhere unless there's a clear use case for setting the host too 🤔

Reply:
There is no specific use case, I just used them as it was the more flexible approach and those where the arguments that had to be passed on. Changed it to port numbers only

Reply:
both the metric and health bind address flags are passed-into/consumed by the  k8s-sigs/controller-runtime's Manager [Options](https://github.com/kubernetes-sigs/controller-runtime/blob/v0.17.2/pkg/manager/manager.go#L218), [Options](https://github.com/kubernetes-sigs/controller-runtime/blob/v0.17.2/pkg/manager/manager.go#L223)

the argument towards keeping them as address string is:  Some Security focussed setups would not want to expose the metrics port on all interfaces, instead only on `127.0.0.1:8080` and then front it with a rbac-proxy which requires auth for the prometheus scrapers to be able to reach it.